projects/core/src/occ/adapters/product/converters/occ-product-search-page-normalizer.ts

```diff
@@ -20,11 +29,38 @@ export class OccProductSearchPageNormalizer
       ...target,
       ...(source as any),
     };
+    this.normalizeFacetValues(target);
     if (source.products) {
       target.products = source.products.map(product =>
         this.converterService.convert(product, PRODUCT_NORMALIZER)
       );
     }
     return target;
   }
+
+  /**
+   *
+   * In case there are so-called `topValues` given for the facet values,
+   * we replace the facet values by the topValues, simple because the
+   * values are obsolete.
+   *
+   * `topValues` is a feature in the adaptive search which can limit a large
+   * amount of facet values to a small set (5 by default). As long as the backend
+   * provides all facet values AND topValues, we normalize the data to not bother
+   * the UI with this specific feature.
+   */
+  private normalizeFacetValues(target: ProductSearchPage): void {
+    if (target.facets) {
+      target.facets.map((facet: Facet) => {
+        if ((<any>facet).topValues) {
```

Comment:
Is casting to `any` done because of this TODO: https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/6597/files#diff-f96c078aac3c2be910056b4149a66670R47 ?

Reply:
I think it was a left over or a mistake. dropped it as it's not necessary